ray-operator/main.go

```diff
@@ -208,7 +209,11 @@ func main() {
 		WorkerSidecarContainers: config.WorkerSidecarContainers,
 	}
 	ctx := ctrl.SetupSignalHandler()
-	exitOnError(ray.NewReconciler(ctx, mgr, rayClusterOptions).SetupWithManager(mgr, config.ReconcileConcurrency),
+	ctx = ray.WithClusterConfig(ctx, ray.ClusterConfig{
+		ForcedClusterUpgrade: config.ForcedClusterUpgrade,
+		EnableBatchScheduler: config.EnableBatchScheduler,
+	})
+	exitOnError(ray.NewReconciler(ctx, mgr, rayClusterOptions).SetupWithManager(ctx, mgr, config.ReconcileConcurrency),
```

Comment:
Passing state by values into `context.Context` is considered an anti-pattern by some people given its unstructured nature. 

What about passing the `configapi.Configuration` struct (or fields) instead where it's needed, which is rather the idiomatic approach operators take in that case. It'll also avoid the need to declare another configuration struct like `ray.ClusterConfig`.

Reply:
> Passing state by values into context.Context is considered an anti-pattern by some people given its unstructured nature.

Not sure to what extent it is so. I come from the Knative project where we actually use it this way to propagate values a lot. Knative follows the K8s project practices and I have seen this pattern used in that code base too, for [example](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/csi/csi_util.go#L202).

> What about passing the configapi.Configuration

I can try passing the configuration and see if that is available where we need it and take it from there. In general with any option there are pros and cons.

Reply:
I can see why this may be debated. In essence some people consider contexts should be primarily used for cancellations and propagating termination signals, and only on last resort for storing values. An example of such discussion can be found at https://www.reddit.com/r/golang/comments/10qf73m/context_api_and_best_practices_of_using_it/.

In the particular case of that PR, I still fail to see the benefit of relying on context, over passing the `configapi.Configuration` object as an argument.

Reply:
> I still fail to see the benefit of relying on context

The generic benefit is not having to change the func signature, allowing passing data down the call chain easier in the future even if they are not a good fit for the config object etc.  Of course there are downsides and maybe here it is an early optimization. In any case I will adapt the PR to use an argument.